Software/src/main/java/org/foi/emp/carmanagement/configs/JwtRequestFilter.java

```diff
@@ -0,0 +1,62 @@
+package org.foi.emp.carmanagement.configs;
+
+import io.jsonwebtoken.ExpiredJwtException;
+import org.foi.emp.carmanagement.services.JwtUserDetailsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+public class JwtRequestFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private JwtUserDetailsService jwtUserDetailsService;
+
+    @Autowired
+    private JwtTokenUtil jwtTokenUtil;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain) throws IOException, ServletException
+
+    {
+        final String requestTokenHeader = response.getHeader("Authorization");
+
+        String username = null;
+        String jwtToken = null;
+        if (requestTokenHeader != null && requestTokenHeader.startsWith("Bearer ")) {
```

Comment:
requestTokenHeader is String, as a Best Practice is Java conducts, use `StringUtils.isNotEmpty()`, it will check if string is not null and not empty.

`private static final String BEARER = "Bearer ";`
and use it here as shown in line 32.